JAICore/jaicore-planning/src/main/java/ai/libs/jaicore/planning/hierarchical/problems/stn/STNPlanningDomain.java

```diff
@@ -14,53 +15,60 @@
 	private final Collection<? extends Operation> operations;
 	private final Collection<? extends Method> methods;
 
-	public STNPlanningDomain(Collection<? extends Operation> operations, Collection<? extends Method> methods) {
+	public STNPlanningDomain(final Collection<? extends Operation> operations, final Collection<? extends Method> methods) {
 		super();
 		this.operations = operations;
 		this.methods = methods;
-		assert isValid();
+		this.checkValidity();
 	}
-	
-	public boolean isValid() {
-		return true;
+
+	public void checkValidity() {
```

Comment:
Why do we have this here at all, then?

Reply:
this can be (and is) overwritten by sub-classes to conduct non-trivial checks.